challenges/classes.js

```diff
@@ -1,7 +1,57 @@
 // 1. Copy and paste your prototype in here and refactor into class syntax.
 
+class CuboidMaker {
+  constructor(props) {
+    this.length = props.length;
+    this.width = props.width;
+    this.height = props.height;
+  }
+
+  volume() {
+    return this.length * this.width * this.height;
+  }
+
+  surfaceArea() {
+    return (
+      2 *
+      (this.length * this.width +
+        this.length * this.height +
+        this.width * this.height)
+    );
+  }
+}
+
+const cuboid = new CuboidMaker({
+  length: 4,
+  width: 5,
+  height: 5
+});
+
 // Test your volume and surfaceArea methods by uncommenting the logs below:
-// console.log(cuboid.volume()); // 100
-// console.log(cuboid.surfaceArea()); // 130
+console.log(cuboid.volume()); // 100
+console.log(cuboid.surfaceArea()); // 130
+
+// Stretch Task: Extend the base class CuboidMaker with a sub class called CubeMaker.  Find out the formulas for volume and surface area for cubes and create those methods using the dimension properties from CuboidMaker.  Test your work by logging out your volume and surface area.
+
+class CubeMaker extends CuboidMaker {
+  constructor(props) {
+    super(props);
+  }
+
+  cubeVolume() {
+    return this.length * this.width * this.height;
```

Comment:
there's not really a point to this method as written. CubeMaker already inherits a volume function from CuboidMaker that performs this calculation in the same way. The method that would make sense to give it would be:

volume() {
    return this.length ** 3;
}
because it is a simpler calculation (only need to access one property vs. three to perform it). So the fact that cubes have all equal sides allows you to replace (technically shadow) the volume method from Cuboid with an optimized version for cubes.